File: java/tutorial/main/ch/softappeal/yass/tutorial/generate/GenerateTypeScript.java
package ch.softappeal.yass.tutorial.generate;

import ch.softappeal.yass.generate.TypeScriptGenerator;
import ch.softappeal.yass.tutorial.contract.Config;

import java.util.Map;

public final class GenerateTypeScript {

    public static void main(final String... args) throws Exception {
        new TypeScriptGenerator(
            Config.class.getPackage().getName(),
            Config.CONTRACT_SERIALIZER,
            Config.INITIATOR,
            Config.ACCEPTOR,
            "../../ts/tutorial/contract-include.txt",
            Map.of( // shows how to use a contract external base type
                Integer.class, new TypeScriptGenerator.ExternalDesc("Integer", "IntegerHandler")
            ),
            "build/generated/ts/contract.ts"
        );
    }

}

File: kotlin/tutorial/main/ch/softappeal/yass/tutorial/generate/GenerateTypeScript.kt
package ch.softappeal.yass.tutorial.generate

import ch.softappeal.yass.generate.ts.ExternalDesc
import ch.softappeal.yass.generate.ts.TypeScriptGenerator
import ch.softappeal.yass.tutorial.contract.Acceptor
import ch.softappeal.yass.tutorial.contract.ContractSerializer
import ch.softappeal.yass.tutorial.contract.Initiator
import java.nio.file.Paths

fun main(args: Array<String>) {
    TypeScriptGenerator(
        "ch.softappeal.yass.tutorial.contract",
        ContractSerializer,
        Initiator,
        Acceptor,
        Paths.get("../../ts/tutorial/contract-include.txt"),
        mapOf(Integer::class.java to ExternalDesc("Integer", "IntegerHandler")), // shows how to use a contract external base type
        Paths.get("build/generated/ts/contract.ts")
        // , "protected readonly __TYPE_KEY__!: never;"
    )
}
